lib/Onyx.js

```diff
@@ -908,67 +965,46 @@ function multiSet(data) {
         .catch(error => evictStorageAndRetry(error, multiSet, data));
 }
 
-// Key/value store of Onyx key and arrays of values to merge
-const mergeQueue = {};
-
-/**
- * @private
- * @param {String} key
- * @returns {Boolean}
- */
-function hasPendingMergeForKey(key) {
-    return Boolean(mergeQueue[key]);
-}
-
 /**
  * Given an Onyx key and value this method will combine all queued
  * value updates and return a single value. Merge attempts are
  * batched. They must occur after a single call to get() so we
  * can avoid race conditions.
  *
  * @private
- * @param {String} key
- * @param {*} data
- *
+ * @param {*} existingValue
+ * @param {*} changes
  * @returns {*}
  */
-function applyMerge(key, data) {
-    const mergeValues = mergeQueue[key];
-    if (_.isArray(data) || _.every(mergeValues, _.isArray)) {
-        // Array values will always just concatenate
-        // more items onto the end of the array
-        return _.reduce(mergeValues, (modifiedData, mergeValue) => [
-            ...modifiedData,
-            ...mergeValue,
-        ], data || []);
+function applyMerge(existingValue, changes) {
+    if (_.isUndefined(existingValue) || _.isNull(existingValue)) {
+        return changes;
     }
 
-    if (_.isObject(data) || _.every(mergeValues, _.isObject)) {
-        // Object values are merged one after the other
-        return _.reduce(mergeValues, (modifiedData, mergeValue) => {
-            // lodash adds a small overhead so we don't use it here
-            // eslint-disable-next-line prefer-object-spread, rulesdir/prefer-underscore-method
-            const newData = Object.assign({}, fastMerge(modifiedData, mergeValue));
-
-            // We will also delete any object keys that are undefined or null.
-            // Deleting keys is not supported by AsyncStorage so we do it this way.
-            // Remove all first level keys that are explicitly set to null.
-            return _.omit(newData, (value, finalObjectKey) => _.isNull(mergeValue[finalObjectKey]));
-        }, data || {});
+    if (_.isArray(existingValue) || _.isArray(changes)) {
+        return changes;
+    }
```

Comment:
Why are we no longer concatenating arrays? It doesn't look like we do that in LocalForage so has the behavior changed or am I missing something?

Reply:
Answer in the next question